test/trip_dispatch_test.rb

```diff
@@ -121,5 +122,151 @@ def build_test_dispatcher
         end
       end
     end
+    
+    describe "Find available Driver" do
+      before do
+        @dispatcher = build_test_dispatcher
+        @driver_found = @dispatcher.find_available_driver()
+      end
+      
+      ## WAVE 1-3 TESTS (PRE-INTELLIGENT DRIVER ASSIGNMENT)
+      # it "finds an available driver" do
+      #   expect(@driver_found.name).must_equal "Driver 2"
+      # end
+      
+      # it "finds the first available driver" do 
+      #   expect(@driver_found.name).wont_equal "Driver 3 (no trips)"
+      # end
+      
+      it "The first available driver has the status :AVAILABLE" do
+        expect(@driver_found.status).must_equal :AVAILABLE
+      end
+      
+      it "Reassigns status when driver is chosen" do
+        @dispatcher.request_trip(1)
+        expect(@driver_found.status).must_equal :UNAVAILABLE
+      end
+      
+      it "Raises ArgumentError if no available drivers" do
+        def build_unavailable_test_dispatcher
+          return RideShare::TripDispatcher.new(
+            directory: UNAVAILABLE_TEST_DATA
+          )
+        end
+        dispatcher = build_unavailable_test_dispatcher
+        expect {
+          dispatcher.request_trip(1)
+        }.must_raise ArgumentError
+      end
+    end
+    
+    describe "find find available driver intelligent assignment" do
+      before do
+        @dispatcher = build_test_dispatcher
+      end
+      
+      it "selects first the driver with no previous trips, then the driver with the least recent most-recent trip" do
```

Comment:
This is nicely organized with comments, but you may consider putting each of these expect in different it blocks.